code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -2,104 +2,164 @@
 	throw_speed = 1
 	throw_range = 5
+
 	w_class = ITEM_SIZE_LARGE
 	origin_tech = "{'combat':1}"
 	material = /decl/material/solid/metal/steel
-	var/status = 0
+
+	var/fire_sound = 'sound/weapons/flamethrower.ogg'
+	/// Whether we have an igniter secured (screwdrivered) to us or not
+	var/secured = FALSE
 	var/throw_amount = 100
-	var/lit = 0	//on or off
-	var/operating = 0//cooldown
-	var/turf/previousturf = null
-	var/obj/item/weldingtool/weldtool = null
+	/// on or off
+	var/lit = FALSE
+	/// cooldown
+	var/operating = FALSE
+	var/turf/previous_turf = null
+	var/obj/item/weldingtool/welding_tool = null
 	var/obj/item/assembly/igniter/igniter = null
 	var/obj/item/tank/tank = null
 
+/obj/item/flamethrower/Initialize(ml, material_key, welder)
+	. = ..()
+	if(welder)
+		welding_tool = welder
+		welding_tool.forceMove(src)
+
+	update_icon()
+
+/obj/item/flamethrower/examine(mob/user, distance)
+	. = ..()
+	if(distance <= 1)
+		if(tank)
+			to_chat(user, SPAN_NOTICE("Release pressure is set to [throw_amount] kPa. The tank has about [round(tank.air_contents.return_pressure(), 10)] kPa left in it."))
+		else
+			to_chat(user, SPAN_WARNING("It has no tank installed."))
+		if(igniter)
+			to_chat(user, SPAN_NOTICE("It has \an [igniter] installed."))
+		else
+			to_chat(user, SPAN_WARNING("It has no igniter installed."))
 
 /obj/item/flamethrower/Destroy()
-	QDEL_NULL(weldtool)
+	QDEL_NULL(welding_tool)
 	QDEL_NULL(igniter)
 	QDEL_NULL(tank)
-	. = ..()
+	return ..()
 
 /obj/item/flamethrower/Process()
 	if(!lit)
-		STOP_PROCESSING(SSobj, src)
-	else
-		var/turf/location = get_turf(src)
-		location?.hotspot_expose(700, 2)
+		STOP_PROCESSING(SSprocessing, src)
+		return null
+
+	var/turf/location = loc
+	if(istype(location, /mob))
+		var/mob/M = location
+		if(M.get_active_hand() == src)
+			location = M.loc
+
+	if(isturf(location)) //start a fire if possible
+		location.hotspot_expose(700, 2)
 
 /obj/item/flamethrower/on_update_icon()
-	overlays.Cut()
+	cut_overlays()
+	add_overlay("_[initial(welding_tool.icon_state)]")
+
 	if(igniter)
-		overlays += "+igniter[status]"
+		add_overlay("igniter_[secured]")
+
 	if(tank)
 		if(istype(tank, /obj/item/tank/hydrogen))
-			overlays += "+htank"
+			add_overlay("tank_hydrogen")
 		else
-			overlays += "+ptank"
+			var/mutable_appearance/M = mutable_appearance(src, "tank")
+			M.color = tank.color
+			add_overlay(M)
+
 	if(lit)
-		overlays += "+lit"
-		item_state = "flamethrower_1"
+		add_overlay("lit")
+		set_light(1.4, 2)
 	else
-		item_state = "flamethrower_0"
-	return
+		set_light(0)
 
 /obj/item/flamethrower/afterattack(atom/target, mob/user, proximity)
+	if(proximity)
+		return
+
+	if(!tank)
+		return
+
+	if(tank.air_contents.get_by_flag(XGM_GAS_FUEL) < 1)
+		to_chat(user, SPAN_WARNING("\The [src] doesn't have enough fuel left to throw!"))
+		return
+
 	// Make sure our user is still holding us
 	if(user && user.get_active_hand() == src)
-		if(user.a_intent == I_HELP) //don't shoot if we're on help intent
```

Comment:
Why remove this check?

Reply:
It's pointless, because we have gun safety system which could be used for stuff like this.

Reply:
Ok, but this isn't a gun.

Reply:
I put it back by the way.